### ISSUE TITLE
counters, progress bars, in-view animations/fx, and waypoints in general

### ISSUE THREAD

**Issue by stratboy** [pojome/elementor#3559]
Hi, this is a feature request. I find in general the current waypoints management a little sloppy. Since Elementor can be full of animations, there should be a way to control when the animations trigger. The main problem is that if the previews contens is byte-heavy, say a big image, the effect will be this:

- for a fraction of time, the big element will have no size, causing below elements to be up in the view

- this will trigger the animations, but they'll be in view only for a moment, so they won't be seen by the user.

- soon after a moment or so,  the big element will take space (possibly all the current viewport height) and start loading, and show

- at this point, animantions of below/following elements will be already complete, so when the user will scroll the page, he won't see them. Too bad.

So, some kind of 'watch' event applied to sections could solve the issue. For example a flag 'watch for animations' applied to sections > then all the animations in page should watch and wait that section to load before starting themselves.

Make sense?

**Comment by stratboy** [comment]
Found a temp way to circumvent the problem. Still, some way to take control of elementor's waypoint would much appreciated.

var image_header = $('.image-header');
image_header.css('height','1000px'); // arbitrary height, to push down following content and thus avoid waypoint to trigger animations

``` js
var checker = setInterval(check_header_height, 200); // arbitrary timing

function check_header_height(){
  if(image_header.find('img').height() > 0){ // initially it will be 0, then soon it will take space
    image_header.css('height','auto'); // take following content back
    $(window).resize(); // force waypoint to refresh positioning check and trigger callbacks
    clearInterval(checker); // stop checking img height
  }
}
```

**Comment by bainternet** [comment]
Duplicate of #3560
